src/function/scalar/string/hex.cpp

```diff
@@ -0,0 +1,217 @@
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/common/vector_operations/unary_executor.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/function/scalar/string_functions.hpp"
+
+namespace duckdb {
+
+struct HexStrOperator {
+	template <class INPUT_TYPE, class RESULT_TYPE>
+	static RESULT_TYPE Operation(INPUT_TYPE input, Vector &result) {
+		auto data = input.GetDataUnsafe();
+		auto size = input.GetSize();
+
+		// Allocate empty space
+		auto target = StringVector::EmptyString(result, size * 2);
+		auto output = target.GetDataWriteable();
+
+		for (idx_t i = 0; i < size; ++i) {
+			memcpy(output, &StringUtil::BYTE_TO_HEXS[static_cast<idx_t>(data[i]) * 2], 2);
+			output += 2;
+		}
+
+		target.Finalize();
+		return target;
+	}
+};
+
+struct FromHexOperator {
+	template <class INPUT_TYPE, class RESULT_TYPE>
+	static RESULT_TYPE Operation(INPUT_TYPE input, Vector &result) {
+		auto data = input.GetDataUnsafe();
+		auto size = input.GetSize();
+
+		if (size > NumericLimits<uint32_t>::Maximum()) {
+			throw InvalidInputException("Hexadecimal input length larger than 2^32 are not supported");
+		}
+
+		D_ASSERT(size <= NumericLimits<uint32_t>::Maximum());
+		auto buffer_size = (size + 1) / 2;
+
+		// Allocate empty space
+		auto target = StringVector::EmptyString(result, buffer_size);
+		auto output = target.GetDataWriteable();
+
+		// Add a leading zero
+		// 'fff' is equivalent to '0fff'
+		idx_t i = 0;
+		if (size % 2 != 0) {
+			*output = StringUtil::HEX_TO_DIGIT[static_cast<idx_t>(data[i])];
```

Comment:
Instead of using a 256 byte buffer could we perhaps check the byte ranges, e.g. something like:

```cpp
uint8_t GetHexValue(char c) {
  if  (c >= '0' && c <= '9') {
     return c - '0';
  }
  if  (c >= 'a' && c <= 'f') {
     return c - 'a';
  }
  if  (c >= 'A' && c <= 'F') {
     return c - 'A';
  }
  throw InvalidInputException("Invalid input for hex digit: %s", string(c, 1));
}
```

That will save on binary size and make it so we get error reporting on invalid digits.